### ISSUE TITLE
feature_request(e-mail): e-mail notifications

### ISSUE THREAD

**Issue by Kristinita** [cheng-kang/wildfire#47]
### 1. Summary

It would be nice, if Wildfire will support automatically e-mail notifications.

+ Admins will receive notifications if post any comment;
+ Another user — replies to his comments and mentions to him.

I don't find, how I can do it in [**Wildfire documentation**](https://wildfire.js.org/).

### 2. Argumentation

Site owners must quickly get comments, that:

+ reply to users
+ ban spammers

Another users must automatically get reminders, if reply or mention; they don't must manually update page pending comment.

It a critical feature for comments services as Wildfire.

### 3. Steps to reproduce

[**I setup Firebase database and add Wildfire to my site**](https://wildfire.js.org/#/usage) → I post comment → I reply to comment from different account:

![Reply to comment](https://i.imgur.com/FT9uy6K.png)

Records in Firebase console:

![Firebase console](https://i.imgur.com/uOC9k7s.png)

### 4. Expected behavior

Successful e-mail notification.

### 5. Actual behavior

No e-mail notification.

I try:

+ GMail
+ Yahoo Mail
+ Tutanota

I check `Spam` folder.

Thanks.

**Comment by momocow** [comment]
@Kristinita  There is not yet email support since the logic purely **exists in the front-end** and firebase is **only for storage**.

To view those notification, you can find those notifications in the notification center on your site as the following image.

![image](https://user-images.githubusercontent.com/17778751/53462024-50a7ad80-3a7d-11e9-83e1-46d56eae533f.png)

And there are your notifications!

![image](https://user-images.githubusercontent.com/17778751/53462083-82207900-3a7d-11e9-99aa-876c74d16a65.png)

**Comment by Kristinita** [comment]
@momocow , `status: question`

Thanks, I know about it. Is any method, that user automatically find out that a new notification has appeared in notification center?

Thanks.

**Comment by momocow** [comment]
@Kristinita Apparently, there's no such way for an user to find out that he has new notifications unless he checks it out in the notification center himself. (at least in the master version)

What do you expect that this behavior should be?

---
IMO, I've noticed that when someone successfully posts his/her comment, a message box is shown up as the following image.

![image](https://user-images.githubusercontent.com/17778751/53465859-8c953f80-3a8a-11e9-9df7-db4ea707ae37.png)

Maybe the message box can also be applied to inform a new notification.
However, since the message box will dismiss itself after several seconds, the information still can be ignored by the user.

How about the FaceBook-like badge counter?
![image](https://user-images.githubusercontent.com/17778751/53466138-a84d1580-3a8b-11e9-8fa3-e67f08d854af.png)

**Comment by momocow** [comment]
Maybe you can try the beta version (#37) and make a plugin to implement the expected behavior?

**Comment by Kristinita** [comment]
@momocow , `type: replies`

> How about the FaceBook-like badge counter?

For this site visitors must visit web-page again. They don't have to do this solely that manually check notifications.

> and make a plugin to implement the expected behavior?

I think, automatical notifications — critical feature. This should be the default, not by plugin.

Thanks.

**Comment by momocow** [comment]
> For this site visitors must visit web-page again. They don't have to do this solely that manually check notifications.

Agree, however the function requires a back-end service support, which is currently unavailable.
See https://stackoverflow.com/questions/37642845/sending-an-email-from-the-browser-using-js.

As a workaround, therefore I suggest using the plugin to implement this until official support.

> I think, automatical notifications — critical feature. This should be the default, not by plugin.

I'm not an authority for this and I respect @cheng-kang's plan for the project.

You can still keep this feature request, as I'm also looking forward to get email notification supported!
